app/src/main/java/com/sdp/movemeet/backend/firebase/firestore/FirestoreActivityManager.java

```diff
@@ -26,4 +29,23 @@ public FirestoreActivityManager(FirebaseFirestore db, String collection, Backend
         return super.search(field, value);
     }
 
+    /**
+     * Returns a list of recently added activities, with the specified size
+     * @param size number of recently added activities we want to get
+     * @return a querySnapshot for size number of recently added activities
+     */
+    public Task<QuerySnapshot> getRecentlyAddedActivities(int size) {
+        return db.collection(collection).orderBy("createdAt", Query.Direction.DESCENDING).limit(size).get();
```

Comment:
It seems smart from an information transfer point of view but what happens if many activities are added to the DB between two "calls" to this function? Not all of them will be retrieved so we have a "hole" in the list of activities, right?

Reply:
Good question ! Let's suppose we have a DB of 30 activities, and the local list is empty. On first fetch, the size will be 30 (as in the updateListActivities method, we check the diff size between local list and DB), and this will get all activities. Now, let's suppose meanwhile we have 5 new activites added to the DB. Then, when the user refreshes the map, another diff check is made. Now, the size will be 5 (as the difference from the local list size 30 and DB size 35 is 5). This method will then get the 5 most recent activities, and add them to the local list. I don't see a case where not all activities from DB would be added to the local list.